File: pset6/cash.py
import math

coins = [0.25, 0.10, 0.05, 0.01]

#receiving cash
while True:
    try:
        owed = float(input("change: "))
        if(owed < 0):
            continue
        else:
            break
    except:
        continue

num_coins = 0



#can possibly be simplified with a function if you want to optimise... im lazy
while(owed > 0):

    if(owed/0.25 >= 1):
        temp = math.floor(owed/0.25)
        num_coins += temp
        owed -= temp*0.25
        owed = round(owed, 2)
        continue

    elif(owed/0.10 >= 1):
        temp = math.floor(owed/0.10)
        num_coins += temp
        owed -= temp*0.10
        owed = round(owed, 2)
        continue

    elif(owed/0.05 >= 1):
        temp = math.floor(owed/0.05)
        num_coins += temp
        owed -= temp*0.05
        owed = round(owed, 2)
        continue

    elif(owed/0.01 >= 1):
        temp = math.floor(owed/0.01)
        num_coins += temp
        owed -= temp*0.01
        owed = round(owed, 2)
        continue

def change()


print(num_coins)

File: pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>
#include <ctype.h>

int main(){
    
    float dollars;
    int num_coins = 0;
    do{
        dollars = get_float("Change owed: ");

    }while(dollars < 0);

    int cents = round(dollars * 100);
    
    while(cents > 0){
        
        if(cents/25 >= 1){
            int temp = floor(cents/25);
            num_coins += temp;
            cents -= temp*25;
            continue;
        }
        
        else if(cents/10 >= 1){
            int temp = floor(cents/10);
            num_coins += temp;
            cents -= temp*10;
            continue;
        }
        
        else if(cents/5 >= 1){
            int temp = floor(cents/5);
            num_coins += temp;
            cents -= temp*5;
            continue;
        }
        
        else if(cents/1 >= 1){
            int temp = floor(cents/1);
            num_coins += temp;
            cents -= temp*1;
            continue;
        }
        
        
    }    
    
    printf("%i\n", num_coins);
    return 0;
}
